SilKit/source/silkit.map

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Vector Informatik GmbH
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+
+/* This version script does not apply symbol versioning, it only changes symbol visibility! */
+{
+    global:
+        SilKit_*;
+
+        extern "C++" {
+            "SilKit::Config::ParticipantConfigurationFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)";
```

Comment:
Thanks, looks better now, with the  demangled / human readable declarations. 
I wonder if we have to explicitly unalias the `std::string`, I guess its some binutils limitation?

Reply:
Since `ld` only knowns about symbols and doesn't understand C or C++ headers, it requires all parts relevant for the mangled name, which - unfortunately - is the full template with all relevant template arguments and the actual namespace where the template lives (the ABI namespace `std::__cxx11`).

@MariusBgm Do we support compiling with `libc++` instead of `libstdc++` as the standard library? If yes, then we might need two entries per C++ symbol (due to the inline namespace).

Reply:
Well, we do support compiling against libc++ from source. 
However, this only affects legacy symbols, so users building from source shouldn't be affected. 
I'm ok with only supporting the legacy symbols for our Ubuntu binary deliveries (e.g. glibc/libstdc++ builds)